pkg/git/repository.go

```diff
@@ -385,18 +389,34 @@ var createRepo = func(ctx context.Context, opts *CloneOptions) (string, error) {
 		return "", fmt.Errorf("failed to create the repository, you can try to manually create it before trying again: %w", err)
 	}
 
+	switch providerType {
+	case Azure:
```

Comment:
I am not a fan of this. In my opinion, each provider should be responsible for parsing out their own URLs.

Reply:
I will try in a following PR. I selfishly want to start messing around with the ADO integration 😏

Reply:
Understandable :)